src/StreamingBodyParser/FormUrlencodedParser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace React\Http\StreamingBodyParser;
+
+use Evenement\EventEmitter;
+use Psr\Http\Message\RequestInterface;
+use React\Http\HttpBodyStream;
+
+final class FormUrlencodedParser extends EventEmitter
+{
+    /**
+     * @var RequestInterface
+     */
+    private $request;
+
+    /**
+     * @var HttpBodyStream
+     *
+     * @internal
+     */
+    public $body;
```

Comment:
Nope because of PHP 5.3 here https://github.com/WyriHaximus/http/blob/97ade3db7e4b1fe079cff9a6643e61975e179b59/src/StreamingBodyParser/FormUrlencodedParser.php#L39-L43 or I could make a public `onEnd` method, mark it private and make everything that shouldn't be `public` `private`.